### PR TITLE
v5 Init: switch to @telegraf/client and type fixes

### PR DESCRIPTION
**Pull request by wojpawlik**
## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/telegram.ts [hunk 1]
```diff
@@ -38,32 +35,6 @@ export class Telegram {
     return this.callApi('getFile', { file_id: fileId })
   }
 
-  /**
-   * Get download link to a file.
-   */
-  async getFileLink(fileId: string | tg.File) {
```
Comment (MKRhere) [inline src/telegram.ts]: Are there any blockers to implement `downloadFile()` in its stead? (#1703)

Comment (wojpawlik) [reply src/telegram.ts]: I wanna defer dealing with Client until transformers are being implemented.

Comment (MKRhere) [reply src/telegram.ts]: If `this.#client.token` is still available, and `downloadFile` is not implemented yet - why remove `getFileLink` already?

Also, how do you see it being implemented with transformers?

Comment (wojpawlik) [reply src/telegram.ts]: Both are simply heavily tied to Client. I removed it cuz I want to get to 0 type errors first.

**Comment by MKRhere** [comment]
Tests are broken from the breaking changes, will just merge this and fix other problems in a separate commit.

File: src/telegram.ts [hunk 2]
```diff
@@ -1,10 +1,32 @@
...
 import * as tt from './telegram-types'
-import ApiClient from './core/network/client'
+import { TT } from './telegram-types'
+import { Client, ClientOptions, createApi, type Opts } from '@telegraf/client'
 import { isAbsolute } from 'path'
 import { URL } from 'url'
+import { TelegramError } from './index'
+
+export class Telegram {
+  client: Readonly<Client>
```
Comment (wojpawlik) [inline src/telegram.ts:10]: It must be `#private`, otherwise every middleware can access `ctx.telegram.client.context`.

Comment (wojpawlik) [reply src/telegram.ts:10]: `Telegraf` must store `Client`.

**Pull request merged**